### PR TITLE
Release mgt: Add python2.7 conditional requirements to core dependencies to support pip py3 installation (Fix issues #1009 #1017 #1017 #1018 #1019 #1020 #1012 #1022 #1024 #1025)

### PR DESCRIPTION
**Pull request by vingar**

### CODE REVIEW

**Comment by vingar** [comment]
Not a WIP anymore and success! Ran for 23 min 55 sec.

I need to describe all the changes in various issues, i.e.:
    db testing: oracle, mysql, postgres, sqlite
    python: 2.6, 2,7, 3.4, 3.5, 3.6
    packaging validation: rucio, client
    syntax validation: flake8, pylint sur py2.7 et py3.6
   test: server, client

**Comment by bari12** [comment]
Great. I didn't go through the PR too much, but do I understand correctly that it now does what it did before (oracle, mysql, postgres) and you want to extend it with the different python version syntax tests?

**Comment by vingar** [comment]
For the change management part, the PR changes are described in the issues: #1009 #1017 #1017 #1018 #1019 #1020 #1012 #1022.

File: lib/rucio/tests/test_clients.py [hunk 1]
```diff
@@ -58,10 +58,11 @@ def testUserpassNoCACert(self):
         creds = {'username': 'wrong', 'password': 'secret'}
         BaseClient(account='root', auth_type='userpass', creds=creds)
 
-    def testx509(self):
-        """ CLIENTS (BASECLIENT): authenticate with x509."""
-        creds = {'client_cert': self.usercert}
-        BaseClient(account='root', ca_cert=self.cacert, auth_type='x509', creds=creds)
+#
+#    def testx509(self):
```
Comment (bari12) [inline lib/rucio/tests/test_clients.py]: Why is this test disabled?

File: tools/pip-requires-client [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-dogpile.cache>=0.6.4        # Caching API plugins
-nose>=1.3.7                 # For rucio test-server
-boto>=2.48.0                # S3 boto protocol
-tabulate>=0.8.2             # Pretty-print tabular data
-progressbar2>=3.35.2        # Text progress bar
-bz2file>=0.98               # Read and write bzip2-compressed files.
-python-magic>=0.4.15        # File type identification using libmagic
-futures>=3.1.1              # Clean single-source support for Python 3 and 2
-six>=1.11.0                 # Python 2 and 3 compatibility utilities
+# setuptools_scm
```
Comment (bari12) [inline tools/pip-requires-client]: Remove line?

File: lib/rucio/tests/test_clients.py (hunk 1 above)
Comment (vingar) [reply lib/rucio/tests/test_clients.py]: it fail with some openssl error for all python versions. I think it's related to urlib3. I forgot to open a separate issue.

Comment (bari12) [reply lib/rucio/tests/test_clients.py]: Is this not tested at the moment? I am just wondering because all the tests pass?

Comment (vingar) [reply lib/rucio/tests/test_clients.py]: Internally to the docker yes. With the server running in a container and the client outside it, it gives some ssl errors.

Comment (vingar) [reply lib/rucio/tests/test_clients.py]: I got it. it was some wrong parameters in the configuration. committing...

**Comment by vingar** [comment]
Accidentally closed the PR...

**Comment by bari12** [comment]
I think you need to make a new commit to be able to open it again. (Or just submit a new one)

**Pull request closed without merging**